src/asm/fstack.c

```diff
@@ -133,6 +135,11 @@ void fstk_AddIncludePath(char const *path)
 	includePaths[nbIncPaths++] = str;
 }
 
+void fstk_SetPreIncludeFile(char const *path)
+{
+	preIncludeName = path;
```

Comment:
What if `preIncludeName` is not `NULL`? Should we error or warn?

Reply:
I'd expect `-P a.asm -P b.asm` to include both in that order...

Reply:
No, the API is explicitly that you can only pre-include a single file. If you need multiple, `INCLUDE` them from a file, and `-P` *that*.

Reply:
`make ASMFLAGS="-P foo.asm"` failing because the makefile already includes another file sounds like bad UX...

Reply:
Pre-including multiple files introduces "far-reaching" complexity in the codebase and breaks an implicit contract in object files, neither of which I find acceptable.

`make ASFLAGS=...` can already fail for a *lot* of reasons already, so care already need to be be taken anyway.

Reply:
> Pre-including multiple files introduces "far-reaching" complexity in the codebase and breaks an implicit contract in object files, neither of which I find acceptable.

This sounds like a "don't do that, then" problem. (I doubt you'll use the feature at all, after all.)

> `make ASFLAGS=...` can already fail for a _lot_ of reasons already, so care already need to be be taken anyway.

True, but one of the best use cases for preincludes is defining a lot of things in a file so you can avoid a long laundry list of `ASFLAGS="-Dfoo=bar -Dbaz=quux"`, so this is one of those things that shouldn't be failing — and `-D` is already something you _want_ to be able to stack like that in the command line (so you can have multiple `-D` flags from multiple sources).

Reply:
The codebase in question being RGBDS, so indeed, we're not doing that.

`-Dbar` twice does not stack, so this works just the same.

Reply:
I didn't make it warn because in general when you specify the same option twice it doesn't warn, currently in rgbds nor in gcc. For example, `rgbasm -o foo.o -o bar.o` just outputs to bar.o, no warning.

Reply:
Maybe that should also warn...

Reply:
@aaaaaa123456789 If you expect to supply a `-P` file with `ASFLAGS`, then write your build commands as `rgbasm -P default.inc $(ASFLAGS) ...`, so a specific `-P` file will override the default one. And if you need the defaults *plus* whatever extra your `ASFLAGS` specify, then if you do `ASFLAGS="-P custom.inc"` just also make sure to `INCLUDE "default.inc"` at the top of custom.inc.

Reply:
Yes, I think both should warn. These options being specified twice almost always indicate a bug, and can cause puzzling errors down the line.

Reply:
Yes, I think both should warn. These options being specified twice almost always indicate a bug, and can cause puzzling errors down the line.

Reply:
Can we leave it as-is for now and consider which options (`-P`, `-o`, and others) need warnings added in a separate issue+PR?

Reply:
If we agree that those options should warn when specified multiple times, then I'm OK with doing so separately.